Strategy.java

```diff
@@ -0,0 +1,29 @@
+import java.util.Arrays;
+
+public class Strategy {
+	private double[] weights;
+	
+	/*Constructor*/
+	public Strategy(double[] w){
+		weights = Arrays.copyOf(w, w.length);
+	}
+	
+	/**
+	 * Calculates the total score of the strategy
+	 * which has a particular set of weights for
+	 * all the available features.
+	 * @param state
+	 * @return sigma(weight[i] * score of feature[i])
+	 */
+	public double calculate(State state){
+		FeaturesScore fs = new FeaturesScore(state);
+		double totalScore = 0.0;
+		
+		for (int i = 0; i < fs.getNumberOfFeatures(); ++i){
```

Comment:
Sources say these micro-optimization doesn't seem to matter in modern compilers. They even say the byte code stays the same. Here's the link to one StackOverflow thread: http://stackoverflow.com/questions/4831748/is-i-really-faster-than-i-in-for-loops-in-java

For consistency purpose, we'll just use i++.